### ISSUE TITLE
Urban Wind Field: URock v2023a - Issue with the extend of the created wind raster

### ISSUE THREAD

**Issue by AlexandraStankulova** [UMEP-dev/UMEP#671]
**Describe the bug**
After launching the Urban Wind Field calculation, UMEP succeeds in creating the Wind Field Raster file but it results with wrong extend. It results rotated and some pieces of the requered area of analysis (the one corrispoding to the area of "Raster template to use as output") are missing.

Here I add the Log data:

QGIS version: 3.34.11-Prizren
QGIS code revision: 2904bcec
Qt version: 5.15.13
Python version: 3.12.6
GDAL version: 3.9.2
GEOS version: 3.12.2-CAPI-1.18.2
PROJ version: Rel. 9.4.0, March 1st, 2024
PDAL version: 2.6.3 (git-version: b5523a)
Algorithm started at: 2024-11-20T11:19:31
Algorithm 'Urban Wind Field: URock v2023a' starting…
Input parameters:
{ 'ATTENUATION_FIELD' : '', 'BUILDINGS' : 'D:/University/PHD/02_ICARUS/GIS/QGIS_PROJECT/PreparationData_Wind/BuildingFootprint/N1_BuildingFootprint.shp', 'HEIGHT_FIELD_BUILD' : 'UN_VOL_AV', 'HORIZONTAL_RESOLUTION' : 5, 'INPUT_PROFILE_FILE' : '', 'INPUT_PROFILE_TYPE' : 0, 'INPUT_WIND_DIRECTION' : 142, 'INPUT_WIND_HEIGHT' : 10, 'INPUT_WIND_SPEED' : 0.7, 'LOAD_OUTPUT' : True, 'OUTPUT_FILENAME' : 'urock_output', 'RASTER_OUTPUT' : 'D:/University/PHD/02_ICARUS/GIS/QGIS_PROJECT/PreparationData_Wind/DSM/N1_DSM.tif', 'SAVE_NETCDF' : True, 'SAVE_RASTER' : True, 'SAVE_VECTOR' : True, 'UROCK_OUTPUT' : 'D:\\University\\PHD\\02_ICARUS\\GIS\\QGIS_PROJECT\\OutputData\\URock_Wind Raster\\Wind Raster_9\\N1', 'VEGETATION' : None, 'VEGETATION_CROWN_BASE_HEIGHT' : '', 'VEGETATION_CROWN_TOP_HEIGHT' : '', 'VERTICAL_RESOLUTION' : 5, 'WIND_HEIGHT' : '1.5' }

Writing settings for this model run to specified output folder (Filename: RunInfoURock_YYYY_DOY_HHMM.txt)
Initiating algorithm
Creates an H2GIS Instance and load data
Creates the stacked blocks used as obstacles
Rotates obstacles to the right direction and calculates geometry properties
Roughness zone properties are:

- z0: 0.05299433083208913
- d: 0.1589829924962674
- Hr: 5.238831747832245
- H_ob_max: 15.0
- lambda_f: 0.010115677193492125
Creates the 2D R?ckle zones
Creates the 2D grid
Initializes the 3D grid within R?ckle zones
Deals with building zones superimposition
Deals with vegetation zones superimposition
Initialize the 3D wind in the grid
Rasterize the data
Apply the wind solver equations
Iteration 1 (max 500) - eps = 0.154219 >= 0.0001

Iteration 51 (max 500) - eps = 0.178511 >= 0.0001

Iteration 101 (max 500) - eps = 0.118241 >= 0.0001

Iteration 151 (max 500) - eps = 0.079372 >= 0.0001

Iteration 201 (max 500) - eps = 0.052431 >= 0.0001

Iteration 251 (max 500) - eps = 0.035147 >= 0.0001

Iteration 301 (max 500) - eps = 0.022968 >= 0.0001

Iteration 351 (max 500) - eps = 0.014954 >= 0.0001

Iteration 401 (max 500) - eps = 0.009638 >= 0.0001

Iteration 451 (max 500) - eps = 0.006131 >= 0.0001

Execution completed in 502.90 seconds (8 minutes 23 seconds)
Results:
{'OUTPUT_FILENAME': 'urock_output',
'UROCK_OUTPUT': 'D:\\University\\PHD\\02_ICARUS\\GIS\\QGIS_PROJECT\\OutputData\\URock_Wind '
'Raster\\Wind Raster_9\\N1'}

Loading resulting layers
Algorithm 'Urban Wind Field: URock v2023a' finished

**Desktop (please complete the following information):**
 - OS: Windows
 - QGIS Version: 3.34.11

**Screenshots**
Here screenshot of the DSM and building polygons extend:
![Image](https://github.com/user-attachments/assets/ffd1f90d-e025-4e11-83b4-125431eb2744)

Here screenshot or the output (the Wind Field raster file):
![Image](https://github.com/user-attachments/assets/5eda9636-4fdb-4816-9c77-c917e5101436)

**Comment by j3r3m1** [comment]
Thank you for reporting. Can you please share your data ?

**Comment by AlexandraStankulova** [comment]
Yes, definetely.
Here I attach the data that I have used:

[Data_WindField.zip](https://github.com/user-attachments/files/17850150/Data_WindField.zip)

Thank you for the help.

**Comment by AlexandraStankulova** [comment]
And here I attach also the output raster fille that I obtained:

[urock_outputWS.zip](https://github.com/user-attachments/files/17850175/urock_outputWS.zip)

Thank you in advance.

**Comment by j3r3m1** [comment]
OK I have also faced the error. This is solved in the new version that you can directly download from the github page as a zip.

However, several points need to be highlighted:
- you use an input raster which has the same size as the building footprint. You should have a bigger domain for the building footprint since the buildings outside your input raster might affect the wind in the smaller domain.
- in rural areas such as the one you are processing, it is better to include vegetation
- the vertical resolution of 5 is high in comparison of building height. You can have a try with 3 m and compare with your simulation at 5 meters. If you consider the difference is negligible, then keep 5. Otherwise 3 m is more appropriate (even though it will be longer to run).

Let me know if everything is fine. If so you can close the issue.

**Comment by AlexandraStankulova** [comment]
Thank you so much for the help.
I will definetely try use building footprint with higher extend,include vegetation and try to change the vertical resolution.
I just wanted to ask from where exaclty i can download the new version you were mentioning about. Ans is it the new version of UMEP?

**Comment by j3r3m1** [comment]
In the github code page of the [UMEP-Processing repo](https://github.com/UMEP-dev/UMEP-processing) you have to click on the green button "Code"
![Image](https://github.com/user-attachments/assets/f1e523f0-95d3-4e61-b722-36be16871af0)

Then click on download zip and then in QGIS you can go to plug-in -> Manage and install plugin and then install from zip.

@biglimp what about having an answer like this in the FAQ ?

**Comment by biglimp** [comment]
Yes!. We have this info here: https://umep-docs.readthedocs.io/en/latest/Getting_Started.html#installing-development-release-could-be-unstable. Can put link in FAQ also.

**Comment by j3r3m1** [comment]
OK we can add this link to a question in the FAQ. I can do it.
What about the text in the link itself. It refers to UMEP only while it should also refer to UMEP-Processing for those using the processing toolbox version. I can also add the link in the 2.6 paragraph.

**Comment by biglimp** [comment]
Please do. Thanks!

**Comment by j3r3m1** [comment]
done

**Comment by AlexandraStankulova** [comment]
First of all thank you so much for the help.

I would like to ask one more thing. I am trying to make the UTCI raster map using "Outdoor Thermal Comfort: Spatial Thermal Comfort" for a whole city.
So,  for now I have successfully arrived to produce the Mean Radiant Temperature rasters using the "Outdoor Thermal Comfort: SOLWEIG"
Now the only thing that is missing before launching the "Outdoor Thermal Comfort: Spatial Thermal Comfort" is the wind raster.
The problem is that my PC is not able to produce the Wind raster for the whole city at once. So, I was thinking to separate the city on portions, calculate the Wind Raster for each portion and then merge the Wind Rasters into one big raster. Since, I have produced at once the Mean Radiant Temperature for the whole city and the Wind raster in portiond that have been merged , will the "Outdoor Thermal Comfort: Spatial Thermal Comfort" work?
Thanks in advance.
